File: app/src/main/java/com/temption/myarchitecture/learn/leetcode/BubbleSort.java
package com.temption.myarchitecture.learn.leetcode;

/**
 * @Description:
 * @Author: haoshuaihui
 * @CreateDate: 2021/4/27 15:26
 */
public class BubbleSort {

    public static void main(String[] args) {
        int[] arr = {1, 5, 7, 8, -1, -3};
        bubbleSort(arr);
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + ">");
        }
    }

    public static void bubbleSort(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            for (int j = 0; j < arr.length - 1 - i; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j + 1];
                    arr[j + 1] = arr[j];
                    arr[j] = temp;
                }
            }
        }
    }
}

File: learn_note/src/main/java/com/ablechange/learn/leetcode/BubbleSort.kt
package com.ablechange.learn.leetcode

/**
 * @Description:相邻元素比较,大的放右边  第一轮 [0 , size-1] 第二轮 [0 ,size - 2]]
 * @Author: haoshuaihui
 * @CreateDate: 2021/4/27 15:26
 */

fun main(args: Array<String>) {
    val arr = intArrayOf(1, 5, 7, 8, -1, -3)
    BubbleSort().bubbleSort(arr)
    for (i in arr.indices) {
        print(arr[i].toString() + ">")
    }
}

class BubbleSort {
    fun bubbleSort(arr: IntArray) {
        for (i in arr.indices) {
            for (j in 0 until arr.size - 1 - i) {
                if (arr[j] > arr[j + 1]) {
                    val temp = arr[j + 1]
                    arr[j + 1] = arr[j]
                    arr[j] = temp
                }
            }
        }
    }
}